### PR TITLE
Fix creating NSG rule when access control enabled with IPv6 single stack

### LINKED ISSUE #5366: NSG rule not created when setting IPv6 single stack allowed source range in v1.26 and v1.25
#### What happened:

``` yaml
apiVersion: v1
kind: Service
metadata:
  name: foo
spec:
  type: LoadBalancer
  selector:
    app: foo
  ipFamilies:
  - IPv6
  ipFamilyPolicy: SingleStack
  loadBalancerSourceRanges:
  - fd00::/8
  ports:
  - name: https
    port: 443
    protocol: TCP
    targetPort: 443
```

When applying the above manifest, the NSG rule was not created as expected.

#### What you expected to happen:

An NSG rule for IPv6 was created.

#### Anything else we need to know?:

It's a bug exposed in #5123 that doesn’t properly handle the scenario of a single IPv6 stack.

#### Environment:

- Kubernetes version (use `kubectl version`): v1.26 / v1.25
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by lodrem**
#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:

Fixes #5366

#### Does this PR introduce a user-facing change?

```release-note
Fix creating NSG rule when access control enabled with IPv6 single stack
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by lodrem** [comment]
@lzhecheng Looks like these two IPv6 jobs aren’t configured properly.

File: pkg/provider/azure_loadbalancer.go [hunk 1]
```diff
@@ -2411,7 +2411,14 @@ func (az *Cloud) reconcileSecurityGroup(clusterName string, service *v1.Service,
 		sourceRanges          = accessControl.SourceRanges()
 		allowedServiceTags    = accessControl.AllowedServiceTags()
 		allowedIPRanges       = accessControl.AllowedIPRanges()
-		sourceAddressPrefixes = accessControl.IPV4Sources()
+		sourceAddressPrefixes = func() []string {
+			// AKS v1.26 does not support dual stack, so it returns the single stack source ranges instead.
+			if service.Spec.IPFamilies[0] == v1.IPv4Protocol {
+				return accessControl.IPV4Sources()
+			} else {
```
Comment (nilo19) [inline pkg/provider/azure_loadbalancer.go:2418]: else is not needed here

Comment (lodrem) [reply pkg/provider/azure_loadbalancer.go:2418]: I would skip this for now as it’s just one line. But, I’m with you on reducing nested or deep branch as much as we can.

**Comment by lodrem** [comment]
The CI pipeline for IPv6 is quite unstable and keeps failing, but it worked fine on my local standalone env.

**Comment by lzhecheng** [comment]
The failures are not related to the change. I will handle the e2e issue in this [PR](https://github.com/kubernetes-sigs/cloud-provider-azure/pull/5370)

**Pull request merged**